.editorconfig

```diff
@@ -0,0 +1,51 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# EditorConfig sets project-wide editor defaults: https://EditorConfig.org
+
+# top-most EditorConfig file, stop looking higher in the tree
+root = true
+
+# Default settings can be overidden by editorconfig file in a subdir
+# or by a specific glob later in this file
+[*]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+indent_style = space
+trim_trailing_whitespace = true
+
+# Python
+[*.py]
+indent_size = 4
+
+# Javascript
+[*.{js,json}]
```

Comment:
Do we have `js` files?

Reply:
No. Do you think its a bad idea to specify a default indentation for such extension even if we currently don't have any?

Reply:
We actually do have a tiny bit of js in the repo. Not as standalone files though, there is some javascript embedded in the html template files used for the docs theme. Although those files aren't pure html either as it's actually a jinja2 template for an html file. Not that this is really relevant for this.